src/net.cpp

```diff
@@ -3454,6 +3465,17 @@ bool CConnman::RemoveAddedNode(const std::string& strNode)
     return false;
 }
 
+bool CConnman::AddedNodesContain(const CAddress& addr) const
+{
+    AssertLockNotHeld(m_added_nodes_mutex);
+    const std::string addr_str{addr.ToStringAddr()};
+    const std::string addr_port_str{addr.ToStringAddrPort()};
+    LOCK(m_added_nodes_mutex);
+    return (m_added_node_params.size() < 24 // bound the query to a reasonable limit
+            && std::any_of(m_added_node_params.cbegin(), m_added_node_params.cend(),
+                           [&](const auto& p) { return p.m_added_node == addr_str || p.m_added_node == addr_port_str; }));
```

Comment:
(hit "submit" too early)

nit, non-blocker: I do not think that this `< 24` cap is needed. What is the max number of added nodes expected?

I think it is not worth doing that, but in theory, if this is to be optimized - then we can observe that `addr_port_str` begins with `addr_str`. For example `aaa:8333` and `aaa` and if we are checking if string `aaaa` is equal to either one, the way this is done now is that it would traverse the entire `aaaa` twice whereas it can be done with a single traverse. But please ignore this :)